src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx

```diff
@@ -69,17 +73,25 @@ const EmbeddingContinuousPlot = () => {
       if (!selectedExpression) {
         dispatch(loadGeneExpression(experimentId, [selectedGene.current]));
       }
+      dispatch(loadCellSets(experimentId));
     }
   }, [experimentId]);
 
   // obj is a subset of what default config has and contains only the things we want change
   const updatePlotWithChanges = (obj) => {
     dispatch(updatePlotConfig(plotUuid, obj));
   };
-
-  const generateVegaData = () => ({
+  const filterSamples = () => {
+    if (config.selectedSample === 'All') {
+      return generateVegaData(data);
+    }
+    const cellIds = Array.from(properties[config.selectedSample].cellIds);
+    const filteredData = data.filter((id) => cellIds.includes(data.indexOf(id)));
+    return generateVegaData(filteredData);
+  };
+  const generateVegaData = (vegaData) => ({
     expression: selectedExpression,
-    embedding: _.cloneDeep(data),
+    embedding: _.cloneDeep(vegaData),
   });
```

Comment:
Technically, what you are doing here makes sense, but the code can be made easier to read by simple restructure. What I suggest is that you do is:
- have function `filterSamples` to just returns the data without calling `generateVegaData` in there. Make `filterSamples` function take in the data it needs to filter as a parameter.
- in `generateVegaData` function, call `filterSamples` on `data` and remove the `vegaData` parameter. Then on the line when you are structuring the embedding data (`embedding: _.cloneDeep(vegaData)`) pass the resulting data returned from calling `generateVegaData`.
- on line 153, call `generateVegaData`, rather than call the `filterSamples` function

I think this will make the code a bit cleaner.